src/Microsoft.TemplateEngine.Cli/New3Command.cs

```diff
@@ -390,27 +460,34 @@ private void DisplayTemplateList(bool showAll = false)
 
             foreach (IGrouping<string, ITemplateInfo> grouping in grouped)
             {
-                List<string> languages = new List<string>();
+                List<string> languageForDisplay = new List<string>();
+                HashSet<string> uniqueLanguages = new HashSet<string>();
 
                 foreach (ITemplateInfo info in grouping)
                 {
                     if (info.Tags != null && info.Tags.TryGetValue("language", out ICacheTag languageTag))
                     {
                         foreach (string lang in languageTag.ChoicesAndDescriptions.Keys)
                         {
-                            if (string.IsNullOrEmpty(Language) && string.Equals(defaultLanguage, lang, StringComparison.OrdinalIgnoreCase))
+                            if (!uniqueLanguages.Contains(lang))
```

Comment:
Couldn't this be converted to `if(uniqueLanuages.Add(lang)` and the call to add at the end of the block be removed?